spec/lib/msf/ui/console/command_dispatcher/exploit_spec.rb

```diff
@@ -1,30 +1,451 @@
+        pending('cmd_check does not support inline values yet')
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check('RHOSTS=192.0.2.5', 'FloatValue=10.0')
+        expected_output = [
+          'Checking for target 192.0.2.5:3000 with normalized datastore value 10.0',
+          'Cleanup for target 192.0.2.5:3000',
+          '192.0.2.5:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.5:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'supports multiple inlined RHOST values' do
+        pending('pending as inline module options are evaluated too late, and the module is therefore treated as a scanner')
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check('RHOSTS=192.0.2.5 192.0.2.6', 'FloatValue=10.0')
+        expected_output = [
+          'Checking for target 192.0.2.5:3000 with normalized datastore value 10.0',
+          'Cleanup for target 192.0.2.5:3000',
+          '192.0.2.5:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.6:3000 with normalized datastore value 5.0',
+          'Cleanup for target 192.0.2.6:3000',
+          '192.0.2.6:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.5 192.0.2.6:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'ignores the -j flag, and the module is not run as a job' do
+        current_mod.datastore['RHOSTS'] = '192.0.2.1'
+        subject.cmd_check('-j')
+        expected_output = [
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 3.5',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+    end
+
+    context 'when checking a non remote exploit module' do
+      let(:current_mod) { non_remote_exploit_mod }
+
+      it 'notifies the user that this module does not support check' do
+        current_mod.datastore['FloatValue'] = nil
+        subject.cmd_check
+        expected_output = [
+          'Check failed: NoMethodError This module does not support check.'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+    end
   end
 
-  describe "#cmd_reload" do
+  describe '#cmd_run' do
+    before do
+      set_default_payload(current_mod)
+    end
+
+    context 'when running a remote exploit module' do
+      let(:current_mod) { remote_exploit_mod }
+
+      it 'reports missing RHOST values' do
+        allow(current_mod).to receive(:run).and_call_original
+        current_mod.datastore['RHOSTS'] = nil
+        subject.cmd_run
+        expected_output = [
+          'Exploit failed: One or more options failed to validate: RHOSTS.',
+          'Exploit completed, but no session was created.'
```

Comment:
Huh, weird I would have expected this message not to appear if something failed as early on as a datastore option

Reply:
Indeed; Fixed in the rhosts pr:
```
      it 'reports missing RHOST values' do
        allow(current_mod).to receive(:run).and_call_original
        current_mod.datastore['RHOSTS'] = nil
        subject.cmd_run
        expected_output = [
          'Msf::OptionValidateError The following options failed to validate: RHOSTS'
        ]

        expect(@combined_output).to match_array(expected_output)
        expect(subject.mod).not_to have_received(:run)
      end
```